### PR TITLE
UCP/GTEST: Conditionally disable slow RMA tests

### PR DESCRIPTION
**Pull request by dmitrygladkov**
## What

Disable some UCP RMA tests

## Why ?

They are disabled due to sporadic slowness - as a result, failure in Jenkins (watchdog kills the test)

## How ?

Check transport and variant for the RMA test

### CODE REVIEW

**Comment by shamisp** [comment]
Do we have issue ticket for this ?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9461/ for details (Mellanox internal link).

File: test/gtest/ucp/test_ucp_rma.cc [hunk 1]
```diff
@@ -89,6 +89,14 @@ class test_ucp_rma : public test_ucp_memheap {
     }
 
     void test_message_sizes(blocking_send_func_t func, size_t *msizes, int iters, int is_nbi);
+
+    void conditionally_disable_test() {
+        // TODO: need to investigate the slownees of the disabled tests
+        if ((GetParam().transports.front().compare("dc_x") == 0) &&
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_rma.cc]: can we add this to init() function? to avoid changing all test functions

**Comment by dmitrygladkov** [comment]
> Why did you add it to many tests? Was not the only one hanging?

I checked e-mail for the messages that were generated by Jenkins and send to me in case of watchdog kills the test:

- nonblocking_stream_get_nb_flush_ep
```
==== backtrace (tid:  49238) ====
 0 0x00000000000dce47 __GI___sched_yield()  :0
 1 0x00000000007738bb ucp_test::progress()  /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node2/worker/2/contrib/../test/gtest/ucp/ucp_test.cc:136
 2 0x00000000006c3998 test_ucp_rma::nonblocking_get_nb()  /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node2/worker/2/contrib/../test/gtest/ucp/test_ucp_rma.cc:87
 3 0x00000000006a4d93 test_ucp_memheap::test_nonblocking_implicit_stream_xfer()  /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node2/worker/2/contrib/../test/gtest/ucp/test_ucp_memheap.cc:106
 4 0x00000000006bfb57 test_ucp_rma_nonblocking_stream_get_nb_flush_ep_Test::test_body()  /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node2/worker/2/contrib/../test/gtest/ucp/test_ucp_rma.cc:313
 5 0x0000000000554d1e ucs::test_base::run()  /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/test.cc:262
 6 0x0000000000544203 testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/gtest-all.cc:3562
 7 0x000000000053867d testing::Test::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/gtest-all.cc:3635
 8 0x000000000053874c testing::TestInfo::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/gtest-all.cc:3812
 9 0x00000000005388af testing::TestCase::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/gtest-all.cc:3930
10 0x000000000053d247 testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/gtest-all.cc:5802
11 0x000000000053d54b testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/gtest-all.cc:5719
12 0x00000000004e5f43 main()  /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/gtest.h:20059
13 0x0000000000021c05 __libc_start_main()  ???:0
14 0x0000000000522e0e _start()  ???:0
=================================
```
- nonblocking_stream_put_nbi_flush_ep
```
==== backtrace (tid:   2436) ====
 0 0x00000000000dce47 __GI___sched_yield()  :0
 1 0x000000000076c59b ucp_test::progress()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/3/contrib/../test/gtest/ucp/ucp_test.cc:136
 2 0x000000000069daae test_ucp_memheap::test_nonblocking_implicit_stream_xfer()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/3/contrib/../test/gtest/ucp/test_ucp_memheap.cc:113
 3 0x00000000006b85f7 test_ucp_rma_nonblocking_stream_put_nbi_flush_ep_Test::test_body()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/3/contrib/../test/gtest/ucp/test_ucp_rma.cc:205
 4 0x0000000000554d1e ucs::test_base::run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/test.cc:262
 5 0x0000000000544203 testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/gtest-all.cc:3562
 6 0x000000000053867d testing::Test::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/gtest-all.cc:3635
 7 0x000000000053874c testing::TestInfo::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/gtest-all.cc:3812
 8 0x00000000005388af testing::TestCase::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/gtest-all.cc:3930
 9 0x000000000053d247 testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/gtest-all.cc:5802
10 0x000000000053d54b testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/gtest-all.cc:5719
11 0x00000000004e5f43 main()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/gtest.h:20059
12 0x0000000000021c05 __libc_start_main()  ???:0
13 0x0000000000522e0e _start()  ???:0
=================================
```
- nonblocking_stream_put_nbi_flush_ep
```
==== backtrace (tid:  30463) ====
 0 0x00000000000dce47 __GI___sched_yield()  :0
 1 0x000000000076c59b ucp_test::progress()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node2/worker/2/contrib/../test/gtest/ucp/ucp_test.cc:136
 2 0x000000000069daae test_ucp_memheap::test_nonblocking_implicit_stream_xfer()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node2/worker/2/contrib/../test/gtest/ucp/test_ucp_memheap.cc:113
 3 0x00000000006b85f7 test_ucp_rma_nonblocking_stream_put_nbi_flush_ep_Test::test_body()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node2/worker/2/contrib/../test/gtest/ucp/test_ucp_rma.cc:205
 4 0x0000000000554d1e ucs::test_base::run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/test.cc:262
 5 0x0000000000544203 testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/gtest-all.cc:3562
 6 0x000000000053867d testing::Test::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/gtest-all.cc:3635
 7 0x000000000053874c testing::TestInfo::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/gtest-all.cc:3812
 8 0x00000000005388af testing::TestCase::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/gtest-all.cc:3930
 9 0x000000000053d247 testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/gtest-all.cc:5802
10 0x000000000053d54b testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/gtest-all.cc:5719
11 0x00000000004e5f43 main()  /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node2/worker/2/contrib/../test/gtest/common/gtest.h:20059
12 0x0000000000021c05 __libc_start_main()  ???:0
13 0x0000000000522e0e _start()  ???:0
=================================
```
- and etc

**Comment by brminich** [comment]
aa, ok. Then would need to move check to init(), as @yosefe suggested

**Comment by dmitrygladkov** [comment]
> aa, ok. Then would need to move check to init(), as @yosefe suggested

sure, working on it right now

File: test/gtest/ucp/test_ucp_rma.cc (hunk 1 above)
Comment (dmitrygladkov) [reply test/gtest/ucp/test_ucp_rma.cc]: good catch! done

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9472/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
@brminich @yosefe could you take a look plz?

**Comment by dmitrygladkov** [comment]
> maybe disable the whole suite then?

Yes, good catch. I thought about this, because I see that those tests take loo long from time to time, but less than already disabled tests

**Comment by dmitrygladkov** [comment]
> > maybe disable the whole suite then?
> 
> Yes, good catch. I thought about this, because I see that those tests take loo long from time to time, but less than already disabled tests
> > Since many tests are problematic (no a single one), looks like other tests could provoke the same slowness as well

@brminich done

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9481/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9482/ for details (Mellanox internal link).

**Pull request merged**